client/src/components/applicants/Applicants.js

```diff
@@ -45,23 +45,24 @@ export class Applicants extends Component {
     copyStages(stages)
   }
 
-  onDrag = (event, applicant) => {
-    if (!this.props.adminView) {
-      event.dataTransfer.setData('text', event.target.id)
-      this.setState({
-        selectedApplicant: applicant
-      })
-    }
-  }
+  onDrop = async (result) => {
+    const { destination, source, draggableId } = result
 
-  onDrop = (stage) => {
-    if (!this.props.adminView) {
-      const { selectedApplicant } = this.state
-      this.props.moveApplicant(selectedApplicant, stage)
-      this.setState({
-        selectedApplicant: ''
-      })
+    if (!destination) { return }
+
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return
     }
+
+    await this.props.moveApplicant(
+      parseInt(draggableId, 10),
```

Comment:
I'm not sure if this needs to be async await, doesn't it just fire away the action and not wait for anything?